### PR TITLE
Bleeding edge base

### PR DESCRIPTION
**Pull request by jroweboy**
Contains the following major changes
- Appveyor uploads to github releases
- Appveyor compiles with mingw
- Appveyor generates an installer using squirrel
- Updater library that uses squirrel to install updates
- Few minor fixes to make the application seem more "legit" to windows (such as rc changes and other things.)
- Updated to the new citra icon

Moved the code to this PR instead so that we have a better reference for what this PR contains exactly. Slowly, I'm getting the major changes in this PR merged into main citra as I clean them up for review.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -110,7 +110,8 @@ else()
     # /O2 - Optimization level 2
     # /GS- - No stack buffer overflow checks
     # /MD - Multi-threaded runtime DLL
-    set(CMAKE_C_FLAGS_RELEASE   "/O2 /GS- /MD" CACHE STRING "" FORCE)
+    # /Zo - enahnced debug info for optimized builds
```
Comment (Dragios) [inline CMakeLists.txt]: `enhanced`

**Comment by JayFoxRox** [comment]
Please seperate this into different feature-commits so it's easier to fix stuff.
At least put the huge (but simple?) changes into a seperate commit so this can be reviewed per commit.

**Comment by JayFoxRox** [comment]
Can we move this PR to this repo so others have a chance to fix it? It's necessary anyway so everybody who'll ever clone this branch will probably want it anyway.

It's conflicting currently (probably due to my change in the mingw-cc PR)

**Comment by jroweboy** [comment]
@JayFoxRox was there not a "fix conflicts" tool in github? I just used it to fix the merge conflict in a few seconds

File: src/updater/updater.cpp [hunk 2]
```diff
@@ -4,12 +4,12 @@
 #include <locale>
 #include <string>
 #include <commctrl.h>
+// clang-format off
```
Comment (JayFoxRox) [inline src/updater/updater.cpp]: We do this by seperating the windows header using an extra comment. `grep` the other files to see how

**Pull request closed without merging**